protostar/upgrader/upgrade_remote_checker.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+
+import requests
+
+from protostar.utils.protostar_directory import VersionManager, VersionType
+
+
+class UpgradeRemoteChecker:
+    """
+    Call a remote endpoint to figure out if the new Protostar version is available.
+    """
+
+    PROTOSTAR_REPO = "https://github.com/software-mansion/protostar"
+
+    @dataclass
+    class Result:
+        latest_release_tag: str
+        latest_version: VersionType
+        is_newer_version_available: bool
+        changelog_url: str
+
+    def __init__(self, version_manager: VersionManager) -> None:
+        self._version_manager = version_manager
+
+    def check(self) -> "UpgradeRemoteChecker.Result":
+        headers = {"Accept": "application/json"}
+        response = requests.get(
+            f"{UpgradeRemoteChecker.PROTOSTAR_REPO}/releases/latest", headers=headers
+        )
+        response_dict = response.json()
+        latest_release_tag = response_dict["tag_name"]
+        latest_version = VersionManager.parse(latest_release_tag)
+        changelog_url = "https://github.com" + response_dict["update_url"]
+        return UpgradeRemoteChecker.Result(
+            latest_version=latest_version,
+            latest_release_tag=latest_release_tag,
+            is_newer_version_available=latest_version
+            > (
+                self._version_manager.protostar_version or VersionManager.parse("0.0.0")
+            ),
+            changelog_url=changelog_url,
+        )
```

Comment:
1. This can be a plain function IMHO
2. Would it make sense to make this `async`?

Reply:
1.  Classes that would use such a function would have to provide `VersionManager` and they would have to pass it to that function, which shouldn't be a responsibility of these classes. This is problematic when dependencies changes. 
2. It would.